labelbox/data/metrics/iou.py

```diff
@@ -0,0 +1,290 @@
+    if tool_type == NDMask:
+        return mask_iou(predictions, labels)
+    elif tool_type in NDTool.get_union_types():
+        return vector_iou(predictions,
+                          labels,
+                          include_subclasses=include_subclasses)
+    elif tool_type in NDClassification.get_union_types():
+        return classification_iou(predictions, labels)
+    else:
+        raise ValueError(f"Unexpected annotation found. Found {tool_type}")
+
+
+def preprocess_args(
+    label_content: List[Dict[str, Any]],
+    ndjsons: List[Dict[str, Any]],
+    include_classifications=True
+) -> Tuple[Dict[str, List[NDAnnotation]], Dict[str, List[NDAnnotation]],
+           List[str]]:
+    """
+
+    This function takes in the raw json payloads, validates, and converts to python objects.
+    In the future datarow_miou will directly take the objects as args.
+
+    Args:
+        label_content : one row from the bulk label export - `project.export_labels()`
+        ndjsons: Model predictions in the ndjson format specified here (https://docs.labelbox.com/data-model/en/index-en#annotations)
+    Returns a tuple containing:
+        - a dict for looking up a list of predictions by feature schema id
+        - a dict for looking up a list of labels by feature schema id
+        - a list of a all feature schema ids
+
+    """
+    labels = label_content['Label'].get('objects')
+    if include_classifications:
+        labels += label_content['Label'].get('classifications')
+
+    predictions = [NDAnnotation(**pred.copy()) for pred in ndjsons]
+
+    unique_datarows = {pred.dataRow.id for pred in predictions}
+    if len(unique_datarows):
+        # Empty set of annotations is valid (if labels exist but no inferences then iou will be 0.)
+        if unique_datarows != {label_content['DataRow ID']}:
+            raise ValueError(
+                f"There should only be one datarow passed to the datarow_miou function. Found {unique_datarows}"
+            )
+
+    labels = [
+        label_to_ndannotation(label, label_content['DataRow ID'])
+        for label in labels
+    ]
+
+    labels = create_schema_lookup(labels)
+    predictions = create_schema_lookup(predictions)
+
+    feature_schemas = set(predictions.keys()).union(set(labels.keys()))
+    return predictions, labels, feature_schemas
+
+
+def datarow_miou(label_content: List[Dict[str, Any]],
+                 ndjsons: List[Dict[str, Any]],
+                 include_classifications=True,
+                 include_subclasses=True) -> float:
+    """
+
+    Args:
+        label_content : one row from the bulk label export - `project.export_labels()`
+        ndjsons: Model predictions in the ndjson format specified here (https://docs.labelbox.com/data-model/en/index-en#annotations)
+        include_classifications: Whether or not to factor top level classifications into the iou score.
+        include_subclassifications: Whether or not to factor in subclassifications into the iou score
```

Comment:
Are `include_classifications`, `include_subclassifications` independent, or should there be some validation on their relation?

Reply:
They are independent.